lab_3/main.py

```diff
@@ -20,7 +21,37 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+
+    if not isinstance(text, str) or not text:
+        return ()
+
+    sentences = re.compile(r'[.|!|?]')
+    sentences = filter(lambda t: t, [t.strip() for t in sentences.split(text)])
+    result = []
+
+    invaluable_trash = ['`', '~', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+',
+                        '=', '{', '[', ']', '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '.', '?', '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+
+    for sentence in sentences:
+        sentence = sentence.lower()
+        for symbols in invaluable_trash:
+            sentence = sentence.replace(symbols, '')
+        tokens = sentence.split()
+        for word in tokens:
+            for letter in word:
+                letter.replace('ö', 'oe')
+                letter.replace('ü', 'ue')
+                letter.replace('ä', 'ae')
+                letter.replace('ß', 'ss')
+        list_of_sentences = []
+        for token in tokens:
+            list_of_sentences.append(tuple(['_'] + list(token) + ['_']))
+        if list_of_sentences == [] or len(sentence) == 1:
```

Comment:
`list_of_sentences == []` can be done without explicit comparison.

Reply:
I have replaced this part of string:
list_of_sentences == [] 


with this one:
if not list_of_sentences